### ISSUE TITLE
Pack loses the names of buildpacks

### ISSUE THREAD

**Issue by ekcasey** [buildpacks/pack#1217]
### Summary
1. `pack buildpack inspect` does not print the names of component buildpacks.
2. `pack builder inspect` does not print the names of any buildpacks
3. When a multibuildpack is packaged only the name of the top-level buildpack in recorded in the `io.buildpacks.buildpack.layers` label
4. When a builder is packaged none of the buildpack names are recorded in the `io.buildpacks.buildpack.layers`

---

##### Steps

1. Observe that both the [Paketo Java Buildpack](https://github.com/paketo-buildpacks/java/blob/main/buildpack.toml#L19) and one of it's children, the [Paketo JMX Buildpack](https://github.com/paketo-buildpacks/jmx/blob/main/buildpack.toml#L19), have `name`s in their respective `buildpack.toml` files
2. Observe that the Paketo JVM Buildpackage contains a name
```
> pack buildpack inspect gcr.io/paketo-buildpacks/jmx
Inspecting buildpack: gcr.io/paketo-buildpacks/jmx
...
Buildpacks:
  ID                           NAME                        VERSION        HOMEPAGE
  paketo-buildpacks/jmx        Paketo JMX Buildpack        3.1.2          https://github.com/paketo-buildpacks/jmx
```
```
> docker pull gcr.io/paketo-buildpacks/jmx
> docker inspect gcr.io/paketo-buildpacks/jmx | jq -r '.[0].Config.Labels."io.buildpacks.buildpack.layers"' | jq -r
{
  "paketo-buildpacks/jmx": {
    "3.1.2": {
      "api": "0.6",
      "stacks": [
        {
          "id": "io.buildpacks.stacks.bionic"
        },
        {
          "id": "org.cloudfoundry.stacks.cflinuxfs3"
        }
      ],
      "layerDiffID": "sha256:c332cab6dd00a6907d9d16a7189340f79a58b09cc3003dda12d70a7e80427231",
      "homepage": "https://github.com/paketo-buildpacks/jmx",
      "name": "Paketo JMX Buildpack"
    }
  }
}
```
3. Observe that the name is lost when this buildpack in included in the Paketo Java Buildpack
```
> pack buildpack inspect gcr.io/paketo-buildpacks/java
Inspecting buildpack: gcr.io/paketo-buildpacks/java
...

Buildpacks:
  ID                                                  NAME                         VERSION        HOMEPAGE
...
  paketo-buildpacks/java                              Paketo Java Buildpack        5.9.1          https://github.com/paketo-buildpacks/java
  paketo-buildpacks/jmx                               -                            3.1.2          https://github.com/paketo-buildpacks/jmx
```
```
> docker pull gcr.io/paketo-buildpacks/java
> docker inspect gcr.io/paketo-buildpacks/java | jq -r '.[0].Config.Labels."io.buildpacks.buildpack.layers"' | jq -r
{
...
  "paketo-buildpacks/java": {
    "5.9.1": {
      "api": "0.6",
      "order": [
        {
          "group": [...]
        }
      ],
      "layerDiffID": "sha256:214d0d5507ac54d7fc3b279f40a980714ff8986903405460f48f0c4f6fa81de7",
      "homepage": "https://github.com/paketo-buildpacks/java",
      "name": "Paketo Java Buildpack"
    }
  },
  "paketo-buildpacks/jmx": {
    "3.1.2": {
      "api": "0.6",
      "stacks": [
        {
          "id": "io.buildpacks.stacks.bionic"
        },
        {
          "id": "org.cloudfoundry.stacks.cflinuxfs3"
        }
      ],
      "layerDiffID": "sha256:c332cab6dd00a6907d9d16a7189340f79a58b09cc3003dda12d70a7e80427231",
      "homepage": "https://github.com/paketo-buildpacks/jmx"
    }
  },
  ```
  
  4. Observe that all names are lost when the Java buildpack is included in a bulider
  ```
  > pack builder inspect paketobuildpacks/builder:base
  ...
  Buildpacks:
  ID                                                  NAME        VERSION        HOMEPAGE
  ...
  paketo-buildpacks/java                              -           5.9.1          https://github.com/paketo-buildpacks/java
  paketo-buildpacks/java-native-image                 -           5.4.0          https://github.com/paketo-buildpacks/java-native-image
  paketo-buildpacks/jmx                               -           3.1.2          https://github.com/paketo-buildpacks/jmx
```

##### Current behavior
Names are lost.

##### Expected behavior
Names of all buildpacks are preserved in `io.buildpacks.buildpack.layers` and displayed by the relevant `pack inspect` commands.

---

##### pack info
```
> pack version
0.19.0+git-360dbae.build-2550
```

##### docker info
N/A

**Comment by haliliceylan** [comment]
I am picking this up. Please assign me @aemengo

**Comment by haliliceylan** [comment]
Hi, I have a question for this issue. I inspected to source code then i realized the inspect command is seems fine right now. Actualy these names are not adding to metadata during the creating builder or multi-buildpack. I find to try where metadata is generating but i could not find them. I tried to look "layerDiffID" key into source code. (because it seems a unique name) 

<img width="579" alt="Screen Shot 2021-08-10 at 12 37 03" src="https://user-images.githubusercontent.com/33789758/128844521-47f23bb9-5cef-4c2e-b555-799cb3ae2dbc.png">
<img width="616" alt="Screen Shot 2021-08-10 at 12 37 19" src="https://user-images.githubusercontent.com/33789758/128844568-06f936ae-0a8c-4677-b79c-d3a83bc6d7c9.png">

I'm little bit lost on this issue, I want to find where metadata is generating and add to these names to metadata. After that i will test with docker inspect and pack inspect. Maybe i need to add some code into pack inspect too.
